fxgl-core/src/main/kotlin/com/almasb/fxgl/core/collection/PropertyMap.kt

```diff
@@ -309,6 +312,20 @@ class PropertyMap {
         }
     }
 
+    override fun write(bundle: Bundle) {
+        // Convert to string map
+        this.toStringMap().forEach { key, value ->
+            // write to bundle
+            bundle.put(key, value);
+        }
+    }
+
+    override fun read(bundle: Bundle) {
+        bundle.data.forEach { (key) ->
```

Comment:
bundle.data is a `Map<String, Serializable>`, so `forEach { key, value -> ... }` should be possible, in which case `bundle.get(key)` can be replaced by just `value` and avoid extra value lookups.

Please check the above is true and update if so.

Reply:
`toValue(bundle.get(key))` has been replaced by `toValue(value.toString())`. Using `value` directly caused tests to fail. Still avoids the extra lookup from before.